internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -134,21 +140,26 @@ func (r *Reconciler) reconcileNode(ctx context.Context, s *scope) (ctrl.Result,
 		}
 	}
 
-	_, nodeHadInterruptibleLabel := node.Labels[clusterv1.InterruptibleLabel]
+	_, nodeHadInterruptibleLabel := s.node.Labels[clusterv1.InterruptibleLabel]
 
 	// Reconcile node taints
-	if err := r.patchNode(ctx, remoteClient, node, nodeLabels, nodeAnnotations, machine); err != nil {
-		return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile Node %s", klog.KObj(node))
+	if err := r.patchNode(ctx, remoteClient, s.node, nodeLabels, nodeAnnotations, machine); err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile Node %s", klog.KObj(s.node))
 	}
 	if !nodeHadInterruptibleLabel && interruptible {
 		// If the interruptible label is added to the node then record the event.
 		// Nb. Only record the event if the node previously did not have the label to avoid recording
 		// the event during every reconcile.
-		r.recorder.Event(machine, corev1.EventTypeNormal, "SuccessfulSetInterruptibleNodeLabel", node.Name)
+		r.recorder.Event(machine, corev1.EventTypeNormal, "SuccessfulSetInterruptibleNodeLabel", s.node.Name)
 	}
 
 	// Do the remaining node health checks, then set the node health to true if all checks pass.
```

Comment:
this comment seems at the wrong place now

Reply:
This comment existed before, I think it somehow makes sense due to the call to summarizeNodeConditions and the code that sets old conditions, but don't mind dropping it if you prefer

Reply:
I think the comment still makes sense. It should just go down before l.162

Now it's in front of a section that does something else